src/Sulu/Bundle/TestBundle/Testing/PurgeDatabaseTrait.php

```diff
@@ -34,6 +34,10 @@ protected static function purgeDatabase(): void
             $connection->executeUpdate('SET foreign_key_checks = 0;');
         }
 
+        if ($connection->getDriver() instanceof \Doctrine\DBAL\Driver\PDOPgSql\Driver) {
+            $connection->executeUpdate('SET session_replication_role = "replica";');
```

Comment:
@danrot This solution is working, what do you think of that?

Reply:
What exactly does it do, resp. where have you found that solution? But as long as it is working I am fine with almost everything :see_no_evil:

Reply:
https://postgresqlco.nf/en/doc/param/session_replication_role/ says:

> The intended use of this setting is that logical replication systems set it to replica when they are applying replicated changes. The effect of that will be that triggers and rules (that have not been altered from their default configuration) will not fire on the replica. See the sql-altertable clauses ENABLE TRIGGER and ENABLE RULE for more information.

> Since foreign keys are implemented as triggers, setting this parameter to replica also disables all foreign key checks, which can leave data in an inconsistent state if improperly used.

I found it here https://stackoverflow.com/a/49584660

Reply:
Sounds good to me :+1: